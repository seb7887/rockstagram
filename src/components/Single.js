import styled from 'styled-components';

const SinglePhoto = styled.div``;

const Single = () => (
  <SinglePhoto>
    I'm the single
  </SinglePhoto>
)

export default Single;