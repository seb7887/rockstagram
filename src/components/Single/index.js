import React from 'react';
import { SinglePhoto } from './style';

import Photo from '../Photo';
import Comments from '../Comments';

const Single = props => {
  const i = props.posts.findIndex(
    post => post.code === props.match.params.postId,
  );
  return (
    <SinglePhoto>
      <Photo key={i} i={i} post={props.posts[i]} {...props} />
      <Comments {...props} postId={i} />
    </SinglePhoto>
  );
};

export default Single;
