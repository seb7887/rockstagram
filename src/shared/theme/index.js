const theme = {
  colors: {
    black: '#0b0b0b',
    offwhite: '#ffffff',
    white: 'white',
    lightgrey: '#fafafa',
    red: 'red',
  },
  fonts: {
    // eslint-disable-next-line quotes
    header: `billabong, 'billabongregular'`,
  },
};

export default theme;
