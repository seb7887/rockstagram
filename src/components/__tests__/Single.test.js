/* eslint-env jest */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from 'react-testing-library';

import Single from '../Single';
import theme from '../../shared/theme';

const removeComment = jest.fn();
const addComment = jest.fn();
const incrementLikes = jest.fn();

const comments = {
  test: [
    {
      text: 'This is a test',
      user: 'kingmob',
    },
  ],
};

const post = {
  caption: 'This is a test',
  code: 'test',
  display_src: 'test.jpg',
  id: '7',
  likes: 7,
};

const posts = [post];

const mockedProps = {
  match: {
    params: {
      postId: 'test',
    },
  },
  comments,
  posts,
  incrementLikes,
  removeComment,
  addComment,
};

describe('<Single/>', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <Router>
        <ThemeProvider theme={theme}>
          <Single {...mockedProps} />
        </ThemeProvider>
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
