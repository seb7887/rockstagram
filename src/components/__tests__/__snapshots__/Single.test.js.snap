// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Single/> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="style__SinglePhoto-nsdyyb-0 gUoAhO"
  >
    <figure
      class="style__Figure-cjo9le-0 kWuZVR"
    >
      <div
        class="grid-photo-wrap"
      >
        <a
          href="/view/test"
        >
          <img
            alt="This is a test"
            class="grid-photo"
            src="test.jpg"
          />
        </a>
        <div
          class="LikeAnimation__Animation-sc-1q1izn2-0 fqDDFP"
        >
          <div>
            <span
              class="like LikeAnimation__Likes-sc-1q1izn2-1 kCBfRn"
            >
              7
            </span>
          </div>
        </div>
      </div>
      <figcaption
        data-testid="caption"
      >
        <p>
          This is a test
        </p>
        <div
          class="control-buttons"
        >
          <button
            class="likes"
          >
            <i
              class="fa fa-heart"
              data-testid="likes"
            />
             7
          </button>
          <a
            class="button"
            href="/rockstagram/view/test"
          >
            <span
              class="comment-count"
              data-testid="count"
            >
              <i
                class="fa fa-comment"
              />
               1
            </span>
          </a>
        </div>
      </figcaption>
    </figure>
    <div
      class="style__CommentGrid-sc-10wf6c9-1 uoWPB"
    >
      <div
        class="style__Comment-sc-10wf6c9-0 fgtolA"
        data-testid="comment"
      >
        <p>
          <strong>
            kingmob
          </strong>
          This is a test
          <button
            class="remove-comment"
            data-testid="remove-comment"
          >
            Ã—
          </button>
        </p>
      </div>
      <form
        class="comment-form"
        data-testid="form"
        id="form"
      >
        <input
          data-testid="author"
          name="author"
          placeholder="Author"
          type="text"
        />
        <input
          data-testid="text"
          name="comment"
          placeholder="Comment here..."
          type="text"
        />
        <input
          hidden=""
          type="submit"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
